### ISSUE TITLE
Request: Option to copy file instead of move

### ISSUE THREAD

**Issue by Amleto** [jkwill87/mnamer#144]
Hi,
I would like a config option to choose between copy or move file. Personally would prefer shutil.copy2 so that the metadata and ownership is maintained. What do you think?

**Comment by soenkehahn** [comment]
Personally I would want to be able to tell `mnamer` to hard-link instead of copy (i.e. `cp -l`). Or maybe even symlink (i.e. `ln -s`). So maybe it'd be best to have an option for setting the copying command that'll be used?

**Comment by PPACI** [comment]
I would like very much to see a symlink option.
I can have a check to implement this if you do not have the time for it.

**Comment by AubreyKilian** [comment]
Is anybody working on this (yet)?  I'll take a stab at it.  My python's a bit rusty, but I need a copy-intead-of-move feature, so I'll see what I can do if nobody else is trying?

**Comment by PPACI** [comment]
I'll have a look on my side when I can.
I'll fork the repo and create a pull request.
It might not be perfect on the first try but at least we will progress on this issue 👍

**Comment by WaywardWizard** [comment]
I have been meaning to do this for awhile. I have issued a merge request. Please review.

**Comment by imchairmanm** [comment]
Just wanted to add another voice in support of this features.  Hardlink / softlink support is pretty much the only thing missing from the workflow I'm trying to implement.  Are there any major roadblocks to accepting the PR that @WaywardWizard created?

For more context into how this would be helpful: I'd love to keep my downloaded files in their original context so that they can continue seeding.  Hardlinks would allow me incorporate content into an organized, well-named library structure without interfering with a torrent client's ability to identify the original file content.

Some other media organization tools that support this type of workflow are Radarr, Sonarr, and filebot.  I'd love to use something like mnamer however as it seems like a more focused and lightweight tool (I have no need for RSS feed monitoring and automatic downloading) with much better ergonomics.

**Comment by Prometheus720** [comment]
I fully agree that link support is a key feature for seeding. You could likely just set it up with --hardlink and --softlink as directives and continue to use the existing syntax but require a directory.
